crates/clarirs_core/src/ast/op.rs

```diff
@@ -144,29 +144,72 @@ impl<'c> AstOp<'c> {
+            AstOp::FpToFp(floating_point, sort) => {
+                floating_point.kind().is_float() && matches!(sort, FSort { .. })
+            }
+            AstOp::BvToFpUnsigned(floating_point, sort, rounding_mode) => {
+                floating_point.kind().is_float()
+                    && matches!(sort, FSort { .. })
+                    && matches!(
+                        rounding_mode,
+                        FPRM::NearestTiesToEven
+                            | FPRM::TowardPositive
+                            | FPRM::TowardNegative
+                            | FPRM::TowardZero
+                            | FPRM::NearestTiesToAway
+                    )
+            }
+            AstOp::FpToIEEEBV(floating_point) => floating_point.kind().is_float(),
+            AstOp::FpToUBV(floating_point, _size, rounding_mode) => {
+                floating_point.kind().is_float()
+                    && matches!(
+                        rounding_mode,
+                        FPRM::NearestTiesToEven
+                            | FPRM::TowardPositive
+                            | FPRM::TowardNegative
+                            | FPRM::TowardZero
+                            | FPRM::NearestTiesToAway
+                    )
+            }
+            AstOp::FpToSBV(floating_point, _size, rounding_mode) => {
+                floating_point.kind().is_float()
+                    && matches!(
+                        rounding_mode,
+                        FPRM::NearestTiesToEven
+                            | FPRM::TowardPositive
+                            | FPRM::TowardNegative
+                            | FPRM::TowardZero
+                            | FPRM::NearestTiesToAway
+                    )
+            }
             AstOp::FpNeg(ast, _) | AstOp::FpAbs(ast, _) => ast.kind().is_float(),
             AstOp::FpAdd(lhs, rhs, _)
             | AstOp::FpSub(lhs, rhs, _)
             | AstOp::FpMul(lhs, rhs, _)
             | AstOp::FpDiv(lhs, rhs, _) => lhs.kind().is_float() && rhs.kind().is_float(),
-            AstOp::FpSqrt(_, _) => todo!(),
-            AstOp::FpEq(_, _) => todo!(),
-            AstOp::FpNeq(_, _) => todo!(),
-            AstOp::FpLt(_, _) => todo!(),
-            AstOp::FpLeq(_, _) => todo!(),
-            AstOp::FpGt(_, _) => todo!(),
-            AstOp::FpGeq(_, _) => todo!(),
-            AstOp::FpIsNan(_) => todo!(),
-            AstOp::FpIsInf(_) => todo!(),
+            AstOp::FpSqrt(a, arg_1) => {
+                a.kind().is_float()
+                    && matches!(
+                        arg_1,
+                        FPRM::NearestTiesToEven
+                            | FPRM::TowardPositive
+                            | FPRM::TowardNegative
+                            | FPRM::TowardZero
+                            | FPRM::NearestTiesToAway
+                    )
+            }
+            AstOp::FpEq(a, b) => a.kind().is_float() && b.kind().is_float(),
```

Comment:
Just as a matter of convention, let's try to use `lhs` and `rhs` for standard two-param ops. Also since many of these are quite repetitive, feel free to merge the match arms like is done near the top of the function with Add, Sub, etc.